File: Programmers/주식가격.java
class Solution {
    public int[] solution(int[] prices) {
        int[] result = new int[prices.length];
        for (int i = 0; i < prices.length-1; i++) {
            int check = 1;
            for (int j = i+1; j < prices.length-1; j++) {
                if (prices[i] <= prices[j]) {
                    check++;
                }
                else break;
            }
            result[i] = check;
        }
        result[result.length-1] = 0;
        return result;
    }
}

File: Programmers/주식가격.py
def solution(prices):
    stack = []
    result = [0 for _ in range(len(prices))]
    for i in range(len(prices)):
        if len(stack) != 0 and prices[i] > prices[stack[-1]]:
            stack.append(i)
        else:
            while len(stack) > 0 and prices[stack[-1]] > prices[i]:
                pop = stack.pop()
                result[pop] = i-pop
            stack.append(i)
    while len(stack) > 0:
        pop = stack.pop()
        result[pop] = len(prices)-pop-1
    return result
